lib/order.rb

```diff
@@ -0,0 +1,46 @@
+require_relative 'menu'
+require_relative 'takeaway'
+
+class Order
+
+  attr_accessor :current_order
+  attr_reader :menu, :receipt, :basket
+
+  def initialize(menu = Menu.new)
+    @menu = menu
+    @select = {}
```

Comment:
Does this need to be an instance variable?

Instance variable should only be used when you need to store the state of an object as a whole, and that state has the same shelf life as the object.
If you just need to store information for a short amount of time, use a local variable. If you need to pass information between a method and another method called inside it, use an argument. Using an instance variable that is meant to be “short lived”, just to be used in a single method and none after is a source of bugs.

Are there other instance variables in this method that don't need to be instance variables?